ui/js/component/formField/view.jsx

```diff
@@ -99,6 +121,16 @@ class FormField extends React.PureComponent {
     return this.refs.field.options;
   }
 
+  validate() {
+    if ("regexp" in this.props) {
+      if (!this.getValue().match(this.props.regexp)) {
+        // Handle invalid input here
```

Comment:
Do we want to just display an error and cancel the form submit ourselves? Feels out of sync with how we've been handling form validation up to now (where the form itself does the check and tells the input to show an error). Two different ways of managing error state in the same component.

Reply:
It should use the existing way of rendering the form error here.

I agree that it should not abort a form submit. Really, this should not know about the form at all.

Likely, the right pattern is:

- Forms don't submit when they have a child component with an error.
- Forms can trigger child form fields to validate themselves one last time on submit

I think the validation in the second case should only happen if the value has changed since last validation.